From db40ca4ed67fec2b38bda13dc94c8bd5ec8beb9d Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Mon, 10 Jan 2022 18:12:45 +0100
Subject: [PATCH 0140/1141] arm64: dts: st: Add Ethernet1 and Ethernet2 node
 for stm32mp25

Add Ethernet1 and Ethernet2 nodes.
IP GMAC is based on SNSPS 5.2x compatible with "snps,dwmac-5.10a"

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: Ic30cdf156994a9ea969f127ec9feccdcc44f3c0d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/235039
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/304906
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 49 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 48 +++++++++++++++++++++++++
 2 files changed, 97 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index e9fa7a5cf541..a229e059cdd2 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -405,6 +405,55 @@ sdmmc1: mmc@48220000 {
 				status = "disabled";
 			};
 
+			eth1: eth1@482c0000 {
+				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
+				reg = <0x482c0000 0x4000>;
+				reg-names = "stmmaceth";
+				interrupts-extended = <&intc GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "macirq";
+				clock-names = "stmmaceth",
+					      "mac-clk-tx",
+					      "mac-clk-rx",
+					      "ptp_ref",
+					      "ethstp",
+					      "eth-ck";
+				clocks = <&rcc CK_ETH1_MAC>,
+					 <&rcc CK_ETH1_TX>,
+					 <&rcc CK_ETH1_RX>,
+					 <&rcc CK_KER_ETH1PTP>,
+					 <&rcc CK_ETH1_STP>,
+					 <&rcc CK_KER_ETH1>;
+				st,syscon = <&syscfg 0x3000 0xffffffff>;
+				snps,mixed-burst;
+				snps,pbl = <2>;
+				snps,axi-config = <&stmmac_axi_config_1>;
+				snps,tso;
+				feature-domains = <&rifsc STM32MP25_RIFSC_ETH1_ID>;
+				status = "disabled";
+				snps,mtl-rx-config = <&mtl_rx_setup_1>;
+				snps,mtl-tx-config = <&mtl_tx_setup_1>;
+
+				stmmac_axi_config_1: stmmac-axi-config {
+					snps,wr_osr_lmt = <0x7>;
+					snps,rd_osr_lmt = <0x7>;
+					snps,blen = <0 0 0 0 16 8 4>;
+				};
+
+				mtl_rx_setup_1: rx-queues-config {
+					snps,rx-queues-to-use = <2>;
+					queue0 {};
+					queue1 {};
+				};
+
+				mtl_tx_setup_1: tx-queues-config {
+					snps,tx-queues-to-use = <4>;
+					queue0 {};
+					queue1 {};
+					queue2 {};
+					queue3 {};
+				};
+			};
+
 			usbh: usb@482e0000 {
 				compatible = "st,stm32mp25-usbh";
 				st,syscfg = <&syscfg 0x2420>;
diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index 3cff3c4c1f5b..bb75164b3fee 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -25,6 +25,54 @@ arm-pmu {
 
 	soc@0 {
 		rifsc: rifsc@42080000 {
+			eth2: eth2@482d0000 {
+				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
+				reg = <0x482d0000 0x4000>;
+				reg-names = "stmmaceth";
+				interrupts-extended = <&intc GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "macirq";
+				clock-names = "stmmaceth",
+					      "mac-clk-tx",
+					      "mac-clk-rx",
+					      "ptp_ref",
+					      "ethstp",
+					      "eth-ck";
+				clocks = <&rcc CK_ETH2_MAC>,
+					 <&rcc CK_ETH2_TX>,
+					 <&rcc CK_ETH2_RX>,
+					 <&rcc CK_KER_ETH2PTP>,
+					 <&rcc CK_ETH2_STP>,
+					 <&rcc CK_KER_ETH2>;
+				st,syscon = <&syscfg 0x3400 0xffffffff>;
+				snps,mixed-burst;
+				snps,pbl = <2>;
+				snps,axi-config = <&stmmac_axi_config_2>;
+				snps,tso;
+				feature-domains = <&rifsc STM32MP25_RIFSC_ETH2_ID>;
+				status = "disabled";
+				snps,mtl-rx-config = <&mtl_rx_setup_2>;
+				snps,mtl-tx-config = <&mtl_tx_setup_2>;
+
+				stmmac_axi_config_2: stmmac-axi-config {
+					snps,wr_osr_lmt = <0x7>;
+					snps,rd_osr_lmt = <0x7>;
+					snps,blen = <0 0 0 0 16 8 4>;
+				};
+
+				mtl_rx_setup_2: rx-queues-config {
+					snps,rx-queues-to-use = <2>;
+					queue0 {};
+					queue1 {};
+				};
+
+				mtl_tx_setup_2: tx-queues-config {
+					snps,tx-queues-to-use = <4>;
+					queue0 {};
+					queue1 {};
+					queue2 {};
+					queue3 {};
+				};
+			};
 		};
 	};
 };
-- 
2.39.2

